### ISSUE TITLE
Separate Mapping from the Application layer

### ISSUE THREAD

**Issue by ahmednfwela** [abpframework/abp#16608]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

Currently in order to map DTOs and Domain entities, you have to use the application layer.

The problem with that is `.DBMigrator` depends on the `.Contracts` layer, but you can't use the DTOs there since the mapping is defined in the `.Application` layer.

And adding a dependency to the Application layer might have some unwanted side effects.

### Describe the solution you'd like

A new `.Mapping` Layer can be added that depends on the `.Domain` and the `.Contracts` layers, whose only concern is Mapping domain entities to DTOs.

The `.Application` layer will depend on the new `.Mapping` layer as well.

This will not be a breaking change.

**Comment by gterdem** [comment]
DbMigrator is not depended on to Application.Contracts to use DTOs, it’s for seeding permissions, features etc.

You shouldn’t inject application service to the DbMigrator.
If you have complex database migration process, you should create a new console application that handles it.

**Comment by ahmednfwela** [comment]
I don't have a complex database migration process, I simply want to be able to reuse the defined mappings without being forced to use the application layer, so creating a new console application that ONLY maps DTOs is an overkill I think.

**Comment by gterdem** [comment]
Why do you want to use DTOs in the DbMigrator?

**Comment by ahmednfwela** [comment]
## The Setup

I have a [meilisearch](https://github.com/meilisearch/meilisearch) instance running, which I need to synchronize data with.
I have multiple routes to go with:
1. Serialize the aggregate root to JSON and send it
    - Pros:
        - I can put my `IDataSeedContributor` in the domain layer, no need for this Issue
    - Cons:
        - Aggregate roots might contain sensitive data, and they might break, this is why we create multiple DTOs for finer control over what the user can see
2. Serialize a special DTO and send it
    - Pros:
        - Mitigate the first approach's cons
        - A faster/better search experience, since you can resolve related data, which can be used during search
        - Share the same mapping with the Application layer, since these DTOs can have common properties
    - Cons:
        - The current issue.

## The Synchronization
as mentioned, I need to synchronize the data to my search instance, which can be a lot of data (several megabytes or even gigabytes), this process is split in 2 parts
- Adding an `MeilisearchContributor : IDataSeedContributor` for the large data in the `.DBMigrator`, this makes me able to:
  - do all the heavy lifting before deployment, this includes setting up an index, grapping related data, serialization, etc...
  - ignore it when running tests
- Adding an `MeilisearchSyncHandler : ILocalEventHandler<EntityChangedEventData<...>>` to the `.Application` layer, which synchronizes the latest version of an `AggregateRoot` to the index

---
TL;DR; I need DTOs in my `DbMigrator` for usage with `MeilisearchContributor` to send large data to an external index.

@gterdem
I can also contribute with a PR to fix this issue.
